tests/ConversationTranslatorTests.ts

```diff
@@ -103,3 +500,203 @@ describe("conversation translator constructor tests", () => {
+        const ct = new sdk.ConversationTranslator(audioConfig);
+        objsToClose.push(ct);
+
+        expect(ct.properties).not.toBeUndefined();
+    });
+
+});
+
+describe("conversation translator service tests",  () => {
+
+    test("Join Conversation Translator, invalid conversation code [400027]", (done: jest.DoneCallback) => {
+
+        // tslint:disable-next-line:no-console
+        console.info("Join Conversation Translator, invalid conversation code [400027]");
+
+        const audioConfig  = sdk.AudioConfig.fromDefaultMicrophoneInput();
+        const ct = new sdk.ConversationTranslator(audioConfig);
+        const code: string = "abcde";
+        const lang: string = "en-US";
+        const nickname: string = "Tester";
+
+        if (endpointHost !== "") { ct.properties.setProperty(sdk.PropertyId.ConversationTranslator_Host, endpointHost); }
+        if (speechEndpointHost !== "") { ct.properties.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+
+        objsToClose.push(ct);
+
+        let errorMessage: string;
+
+        ct.joinConversationAsync(code, nickname, lang,
+            undefined,
+            ((error: any) => {
+                expect(error).toContain("400027");
+                errorMessage = error;
+            }));
+
+        WaitForCondition(() => (errorMessage !== undefined), done);
+
+    });
+
+    test("Join Conversation Translator, duplicate nickname [400028]", (done: jest.DoneCallback) => {
+
+        // tslint:disable-next-line:no-console
+        console.info("Join Conversation Translator, duplicate nickname [400028]");
+
+        let errorMessage: string = "";
+
+        // start a conversation
+        const config = sdk.SpeechTranslationConfig.fromSubscription(Settings.SpeechSubscriptionKey, Settings.SpeechRegion);
+        if (endpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.ConversationTranslator_Host], endpointHost); }
+        if (speechEndpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+        config.setProperty(sdk.PropertyId[sdk.PropertyId.ConversationTranslator_Name], "Tester");
+
+        const c = sdk.Conversation.createConversationAsync(config, (() => {
+
+            c.startConversationAsync((() => {
+
+                const ct = new sdk.ConversationTranslator();
+                if (endpointHost !== "") { ct.properties.setProperty(sdk.PropertyId.ConversationTranslator_Host, endpointHost); }
+                if (speechEndpointHost !== "") { ct.properties.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+
+                objsToClose.push(ct);
+
+                const nickname: string = "Tester";
+                const lang: string = "en-US";
+
+                ct.joinConversationAsync(c.conversationId, nickname, lang,
+                    (() => {
+                        done();
```

Comment:
>     done(); [](start = 19, length = 12)

If the error is expected, should this be done.fail()?

Reply:
Removed some of the dones which weren't needed.